runbot_send_email/models/runbot_build.py

```diff
@@ -0,0 +1,167 @@
+            r"([~\w,\-,\_]+)/" + r"([\w,\-,\_]+)(.git){0,1}((/){0,1})"
+        for rec in self:
+            rec.repo_host, rec.repo_owner, rec.repo_project = '', '', ''
+
+            match_object = re.search(repo_git_regex, rec.repo_id.name)
+            if match_object:
+                rec.repo_host = match_object.group(3)
+                rec.repo_owner = match_object.group(5)
+                rec.repo_project = match_object.group(6)
+            rec.repo_link = "https://" + rec.repo_host + '/' + rec.repo_owner \
+                + '/' + rec.repo_project
+            rec.pr_link = rec.repo_link + rec.branch_id.name.replace(
+                'refs/heads', '/tree').replace('refs', '')
+            if (hasattr(rec.branch_id.repo_id, 'uses_gitlab') and
+                    rec.branch_id.repo_id.uses_gitlab):
+                rec.pr_link = rec.pr_link.replace('/pull/', '/merge_requests/')
+            rec.commit_link = rec.repo_link + '/commit/' + rec.name[:8]
+
+    @api.multi
+    def _host_name(self):
+        base_url = self.env['ir.config_parameter'].get_param('web.base.url')
+        for record in self:
+            record.host_name = urlparse(base_url).hostname
+
+    @api.multi
+    def _branch_name(self):
+        for record in self:
+            if 'pull' in record.branch_id.name:
+                branch = _("%s #{}" % (
+                    'PR' if not (
+                        hasattr(record.branch_id.repo_id, 'uses_gitlab') and
+                        record.branch_id.repo_id.uses_gitlab) else 'MR')
+                )
+                branch = branch.format(record.branch_id.branch_name)
+            else:
+                branch = record.branch_id.branch_name
+            record.branch_name = branch
+
+    @api.multi
+    def _status_build(self):
+        for record in self:
+            status = 'Broken'
+            if record.state == 'testing':
+                status = 'Testing'
+            elif record.state in ('running', 'done'):
+                if record.result == 'ok':
+                    status = 'Fixed'
+                elif record.result == 'warn':
+                    status = "Warning"
+            record.status_build = status
+
+    @api.multi
+    def _subject_email(self):
+        for record in self:
+            pr_reg = "(\\/pull\\/)"
+            match_pr = re.search(pr_reg, record.branch_id.name)
+
+            if match_pr:
+                subject_temp = _("[runbot] {}/{}#{}")\
+                    .format(record.repo_owner, record.repo_project,
+                            record.branch_id.branch_name)
+            else:
+                subject_temp = _("[runbot] {}/{}#{} - {}")\
+                    .format(record.repo_owner, record.repo_project,
+                            record.branch_id.branch_name, record.name[:8])
+
+            record.subject_email = subject_temp
+
+    @api.multi
+    def send_email(self):
+        config_parameters = self.env['ir.config_parameter'].sudo()
```

Comment:
Use template.send_email like as https://github.com/odoo/odoo/blob/11.0/addons/rating/models/rating.py#L218

Reply:
and support multi